### PR TITLE
Added a carousels with cards

### PR DESCRIPTION
**Pull request by Artur0412-KU**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import Smartphones from './pages/Smartphones/Smartphones';
 import Laptops from './pages/Laptops/Laptops';
 import Viewed from './pages/Viewed/Viewed';
 import './styles/globals.css';
-import AppRoute from './enums/enum';
+import AppRoute from './/routes/Route';
```
Comment (mskmee) [inline src/App.tsx]: It's enum, left it in enum folder. Create a sepparete file in enums and reexport from the enums.ts

File: src/componets/Card/BrandCard.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Card } from 'antd';
+import React from 'react';
+import { brandData } from './constants';
+import styles from './card.module.css';
+
+export default function BrandCard() {
+  return (
+    <>
+      {brandData.map((item) => (
```
Comment (mskmee) [inline src/componets/Card/BrandCard.tsx:8]: key

File: src/componets/Card/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import { Card, Rate } from 'antd';
+import styles from './card.module.css';
+import { data } from './constants';
+import { useState } from 'react';
+
+const ProductCard = () => {
+  const [isHovered, setIsHovered] = useState(false);
```
Comment (mskmee) [inline src/componets/Card/Card.tsx]: we can use here css pseudpo class, as Rozy mention it in previous repo. Let's move it to css please

File: src/componets/Card/Card.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  const handleMouseEnter = () => {
+    setIsHovered(true);
+  };
+
+  const handleClickLike = () => {
+    setIsLiked((prevState) => !prevState);
+  };
+  return (
+    <>
+      {data.map((item) => (
```
Comment (mskmee) [inline src/componets/Card/Card.tsx:23]: don't forget about key, seems like when your are replace hover and other logic to the css, all nested items could be moved to the card component or some other card details component

File: src/componets/Card/LaptopCard.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+  const handleMouseEnter = () => {
+    setIsHovered(true);
+  };
+
+  const handleClickLike = () => {
+    setIsLiked((prevState) => !prevState);
+  };
+
+  return (
```
Comment (mskmee) [inline src/componets/Card/LaptopCard.tsx:21]: the same thing, like in previous Card component. And seems like we don't need  two almost the same components.

File: src/componets/Carousel/Carousel.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import { ChildCard } from '../../types/types';
+
+const Carousels = ({ children, className, responsive }: ChildCard) => {
+  const ref = useRef<CarouselRef | null>(null);
+  const [prevClick, setPrevClick] = useState(false);
+  const [nextClick, setNextClick] = useState(false);
+
+  const [isLastSlick, setIsLastSlick] = useState(Boolean);
+  const [isFirstSlick, setIsFirstSlick] = useState(Boolean);
+  // const [currentSlickIndex, setCurrentSlickIndex] = useState(0);
```
Comment (mskmee) [inline src/componets/Carousel/Carousel.tsx]: delete commended code

File: src/interfaces/interfaces.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export interface ResponsiveSettings {
```
Comment (mskmee) [inline src/interfaces/interfaces.ts]: lets avoid interfaces and use types instead.

File: src/types/types.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { ReactNode } from 'react';
+import styles from '../components/Carousels.module.css';
+import { ResponsiveSettings } from '../interfaces/interfaces';
+
+export type TextLink = {
```
Comment (mskmee) [inline src/types/types.ts]: one file - one type

File: src/types/types.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+  text: string;
+  link: string;
+};
+
+export type ChildCard = {
```
Comment (mskmee) [inline src/types/types.ts]: if type related only for one component, then you should describe it in the component file.

**Review by mskmee — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (Artur0412-KU) [reply src/App.tsx]: Why I reexport from the enums.ts, not Routes.ts?

File: src/componets/Card/Card.tsx (hunk 3 above)
Comment (Artur0412-KU) [reply src/componets/Card/Card.tsx]: I used pseudo classes in module css file

File: src/componets/Card/Card.tsx [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+          <div className={styles.cardContainerTop}>
+            <img src={item.img} className={styles.productImage} />
+            <div className={styles.cardConatinerLike}>
+              <img
+                onClick={handleClickLike}
+                src={isLiked ? item.likeIcon : item.likeIconClick}
+                alt="Like"
+                className={styles.likeIcon}
+              />
+              <img src={item.colorPalette} className={styles.colorPalette} />
```
Comment (Rozygeldi94) [inline src/componets/Card/Card.tsx]: you should write an alt attribute for all img tags

Comment (Artur0412-KU) [reply src/componets/Card/Card.tsx]: Added)

**Pull request merged**